app/models/manageiq/providers/microsoft/infra_manager/powershell.rb

```diff
@@ -106,5 +118,16 @@ def run_powershell_script(script)
 
       results
     end
+
+    def with_winrm_shell(shell_type = :powershell)
+      with_provider_connection do |connection|
+        begin
+          shell = connection.shell(shell_type)
+          yield shell
+        ensure
+          shell.close
+        end
+      end
+    end
```

Comment:
@djberg96 can you place both "with_winrm_shell" definitions contiguous in the source file?  Since you're adding both of them I can see no reason to have them at opposite ends of the file.  Otherwise LTGM.